plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryMetadata.java

```diff
@@ -417,6 +448,32 @@ public void dropTable(ConnectorSession session, ConnectorTableHandle tableHandle
         client.dropTable(tableId);
     }
 
+    @Override
+    public ConnectorInsertTableHandle beginInsert(ConnectorSession session, ConnectorTableHandle tableHandle, List<ColumnHandle> columns, RetryMode retryMode)
+    {
+        if (retryMode != RetryMode.NO_RETRIES) {
+            throw new TrinoException(NOT_SUPPORTED, "This connector does not support query retries");
+        }
+        BigQueryTableHandle table = (BigQueryTableHandle) tableHandle;
+        if (isWildcardTable(TableDefinition.Type.valueOf(table.getType()), table.getRemoteTableName().getTableName())) {
+            throw new TrinoException(BIGQUERY_UNSUPPORTED_OPERATION, "This connector does not support inserting into wildcard tables");
+        }
+        ImmutableList.Builder<String> columnNames = ImmutableList.builderWithExpectedSize(columns.size());
+        ImmutableList.Builder<Type> columnTypes = ImmutableList.builderWithExpectedSize(columns.size());
+        for (ColumnHandle columnHandle : columns) {
+            BigQueryColumnHandle column = (BigQueryColumnHandle) columnHandle;
+            columnNames.add(column.getName());
+            columnTypes.add(column.getTrinoType());
+        }
+        return new BigQueryInsertTableHandle(table.getRemoteTableName(), columnNames.build(), columnTypes.build());
```

Comment:
q: It seems following code in BaseJdbcClient is redundant then - https://github.com/trinodb/trino/blob/28a62e1ac5897635cfdf224a434b174a6c274ea4/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java#L619-L620

Because both in BigQuery and BaseJdbcClient we populate the remote names when `getTableHandle` is called.

Or do we need to re-fetch the remote name here?

Reply:
I think we don't need to refresh here.

Reply:
That makes sense to me as well. I'll check why JDBC does it though and send a PR accordingly. Thanks.